[SPARK-5088] Use spark-class for running executors directly

jongyoul opened the pull request.

jongyoul committed:
[SPARK-5088] Use spark-class for running executors directly
- Changed command for using spark-class directly
- Delete sbin/spark-executor and moved some codes into spark-class' case statement

jongyoul committed:
[SPARK-5088] Use spark-class for running executors directly
- Adjusted orders of import

SparkQA commented:
[Test build #25051 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25051/consoleFull) for   PR 3897 at commit [`ed906b7`](https://github.com/apache/spark/commit/ed906b7ae16c08521f8e3b4aa3da47fd293abff3).
 * This patch merges cleanly.

SparkQA commented:
[Test build #25051 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25051/consoleFull) for   PR 3897 at commit [`ed906b7`](https://github.com/apache/spark/commit/ed906b7ae16c08521f8e3b4aa3da47fd293abff3).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `      val executorPath = new File(executorSparkHome, s"/bin/spark-class $executorBackendName")`
  * `      command.setValue(s"cd $`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25051/
Test PASSed.

jongyoul commented:
@tnachen On the contrary, why mesos has two different launcher? I couldn't find any idea in a history of that codes, and spark-executor calls spark-class which has a case statement internally. I think that merging two similar codes is a good idea in maintaining codes.

tnachen commented:
I'm not part of that history either, I just want to know the motivation since the PR doesn't say anything.
For code cleanup it does make sense. 
Your patch looks good to me, unfortunately we need to find some committer to look at this now....

jongyoul commented:
@tnachen I'm sorry for not having any comment. I think that this PR is very clear to guess my opinion.

@pwendell Please review this PR which makes some mesos codes clean. Because this PR is about maintaining code, I ask you to review this PR.

jongyoul commented:
@JoshRosen @tgravescs @andrewor14 Could anyone review this PR? That makes mesos codes clean

jongyoul commented:
/cc @mateiz How about this patch? I don't know who maintains mesos' codes mainly. Because you are one of committers of mesos, I ask you to review this simple PR, which is about removing redundant script file. I've already tested this PR in my cluster

jongyoul commented:
Please review this PR

mateiz commented:
This looks okay, but is there any motivation for it other than getting rid of that source file?

mateiz commented on core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala:
```diff
@@ -118,14 +119,16 @@ private[spark] class MesosSchedulerBackend(
     if (uri == null) {
-      val executorPath = new File(executorSparkHome, "/sbin/spark-executor").getCanonicalPath
+      val executorPath = new File(executorSparkHome, s"/bin/spark-class $executorBackendName")
+        .getCanonicalPath
```
This is actually weird, why are we including `$executorBackendName` in the filename? We should just look for spark-class. I'm not even sure getCanonicalPath will work on a file that doesn't exist

jongyoul replied:
That's my mistake. I'll fix it right now

jongyoul commented:
@mateiz That's all, but I think it is meaningful to remove some redundant codes and helps understand codes easy. And I wonder why that code exists. I think 'spark-executor`is just a wrapper for launching `spark-class`. And we need to try to merge coarse and fine grained mode. That's the first and easiest effort for it

mateiz commented:
Coarse and fine-grained mode are different on purpose, I don't think it makes sense to merge those. BTW have you tested both of them on this?

jongyoul commented:
@mateiz I've test it both mode.

jongyoul committed:
[SPARK-5088] Use spark-class for running executors directly
- Fixed code if spark.executor.uri doesn't have any value
- Added test cases

SparkQA commented:
[Test build #25494 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25494/consoleFull) for   PR 3897 at commit [`b5474ae`](https://github.com/apache/spark/commit/b5474ae73764836a235d3dd6ec923d394b812808).
 * This patch merges cleanly.

jongyoul committed:
[SPARK-5088] Use spark-class for running executors directly
- Rebased from master

Conflicts:
	core/src/test/scala/org/apache/spark/scheduler/mesos/MesosSchedulerBackendSuite.scala

SparkQA commented:
[Test build #25495 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25495/consoleFull) for   PR 3897 at commit [`ddf7da4`](https://github.com/apache/spark/commit/ddf7da43c2066373b16581197e350f795bd6081e).
 * This patch merges cleanly.

jongyoul commented:
@mateiz I've fixed weird codes by your comments and added test case whether `spark.executor.uri` exists or not.

jongyoul commented:
@mateiz Could you please explain the purposes of coarse and fine-grained mode shortly? It looks similar. The only difference between two modes is the way to offer resources. I have no idea of history of existing two modes.

SparkQA commented:
[Test build #25497 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25497/consoleFull) for   PR 3897 at commit [`6b8f169`](https://github.com/apache/spark/commit/6b8f169c1bcf792998304fd4d17d369b4f46126d).
 * This patch merges cleanly.

SparkQA commented:
[Test build #25497 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25497/consoleFull) for   PR 3897 at commit [`6b8f169`](https://github.com/apache/spark/commit/6b8f169c1bcf792998304fd4d17d369b4f46126d).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `      command.setValue(s"cd $`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25497/
Test FAILed.

SparkQA commented:
[Test build #25495 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25495/consoleFull) for   PR 3897 at commit [`ddf7da4`](https://github.com/apache/spark/commit/ddf7da43c2066373b16581197e350f795bd6081e).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `      command.setValue(s"cd $`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25495/
Test PASSed.

jongyoul commented:
retest this please.I don't know why that simple test fails.

SparkQA commented:
[Test build #25502 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25502/consoleFull) for   PR 3897 at commit [`6b8f169`](https://github.com/apache/spark/commit/6b8f169c1bcf792998304fd4d17d369b4f46126d).
 * This patch merges cleanly.

SparkQA commented:
[Test build #25494 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25494/consoleFull) for   PR 3897 at commit [`b5474ae`](https://github.com/apache/spark/commit/b5474ae73764836a235d3dd6ec923d394b812808).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `      command.setValue(s"cd $`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25494/
Test PASSed.

SparkQA commented:
[Test build #25502 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25502/consoleFull) for   PR 3897 at commit [`6b8f169`](https://github.com/apache/spark/commit/6b8f169c1bcf792998304fd4d17d369b4f46126d).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `      command.setValue(s"cd $`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25502/
Test PASSed.

jongyoul commented:
@mateiz Passed tests. Please review it again.

SparkQA commented:
[Test build #25533 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25533/consoleFull) for   PR 3897 at commit [`e3cb711`](https://github.com/apache/spark/commit/e3cb711c1581d9083c98cf84736c49e99a503432).
 * This patch merges cleanly.

SparkQA commented:
[Test build #25533 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25533/consoleFull) for   PR 3897 at commit [`e3cb711`](https://github.com/apache/spark/commit/e3cb711c1581d9083c98cf84736c49e99a503432).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `      command.setValue(s"cd $`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25533/
Test PASSed.

jongyoul commented:
@mateiz Check this PR finally. I think It would be good by your review.

mateiz commented:
@jongyoul sorry for the delayed reply, but this looks good. Mind rebasing it on master though? It's now fallen behind.

jongyoul commented:
@mateiz I'll rebase this PR from master.

SparkQA commented:
[Test build #25732 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25732/consoleFull) for   PR 3897 at commit [`25f3617`](https://github.com/apache/spark/commit/25f3617182c8d4491a0545e3231dbdafe668c4a5).
 * This patch **does not merge cleanly**.

jongyoul commented:
Rebase is not finished.

SparkQA commented:
[Test build #25734 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25734/consoleFull) for   PR 3897 at commit [`932289f`](https://github.com/apache/spark/commit/932289f6d808932da9fa54c21b32c61efca5a18f).
 * This patch merges cleanly.

SparkQA commented:
[Test build #25732 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25732/consoleFull) for   PR 3897 at commit [`25f3617`](https://github.com/apache/spark/commit/25f3617182c8d4491a0545e3231dbdafe668c4a5).
 * This patch **passes all tests**.
 * This patch **does not merge cleanly**.
 * This patch adds the following public classes _(experimental)_:
  * `      command.setValue(s"cd $`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25732/
Test PASSed.

jongyoul commented:
retest this please.

SparkQA commented:
[Test build #25734 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25734/consoleFull) for   PR 3897 at commit [`932289f`](https://github.com/apache/spark/commit/932289f6d808932da9fa54c21b32c61efca5a18f).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25734/
Test FAILed.

jongyoul committed:
[SPARK-5088] Use spark-class for running executors directly
- Added a listenerBus for fixing test cases

SparkQA commented:
[Test build #25738 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25738/consoleFull) for   PR 3897 at commit [`8232aa8`](https://github.com/apache/spark/commit/8232aa8b07a10cb6d1e07e8be49741585f1b4126).
 * This patch merges cleanly.

SparkQA commented:
[Test build #25738 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/25738/consoleFull) for   PR 3897 at commit [`8232aa8`](https://github.com/apache/spark/commit/8232aa8b07a10cb6d1e07e8be49741585f1b4126).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/25738/
Test PASSed.

jongyoul commented:
@mateiz I've rebased this PR and finished tests successfully. Merge this, please.

pwendell commented:
LGTM I will pull it in.

mateiz commented:
Thanks, looks good!
